### ISSUE TITLE
Decide which adjacency reconstruction normalisation to use

### ISSUE THREAD

**Issue by wehlutyk** [ixxi-dante/nw2vec#14]
This is the follow-up to point 2 of question 1 [here](https://github.com/ixxi-dante/nw2vec/issues/3#issuecomment-382755014).

In short, when using the straightforward cross-entropy value for adjacency reconstruction, we give equal weight to neighbours and non-neighbours in the prediction accuracy. Ideally, what we'd like to do is predict, for each node, the set of neighbours and the set of non-neighbours, without penalising our prediction loss by the fact that there are (usually) many more non-neighbours than there are neighbours.

The current solution (in #10, soon to be merged) is to scale the items in the KL loss so that, for a given node, its neighbours contribute 1/2 on average, and its non-neighbours contribute 1/2 on average. We then average over all nodes to get a loss that contributes 1, but weighs neighbours and non-neighbours equally.

Another way would be to upscale true positives and true negatives, and downscale false positives and false negatives.

**Comment by wehlutyk** [comment]
#38 showed that the current solution is much worse than no weighing. so we're going to move on with no weighing until other options are explored.

One question is why the vanilla `gae` does such weighing (see the `norm` and `pos_weight` arguments in https://github.com/tkipf/gae/blob/master/gae/optimizer.py#L28 , and its definition in https://github.com/tkipf/gae/blob/master/gae/train.py#L77-L78), and what it's effect is then.

Tasks here:
- [ ] re-check that the implementations of `gae` and what we had for weighing do match
- [ ] test `gae` on BlogCatalog

**Comment by wehlutyk** [comment]
So #48 fixed some bugs and showed that weighing is not necessarily much worse than not weighing, it seems to mainly change the density of the final reconstruction. I now leave the choice open when creating the model (e46dac078c05e44a57c22046e0f0a4180c08dc3e). The above tasks are still relevant though, and might go with #44..

**Comment by wehlutyk** [comment]
Closing this as #44 and #53 are done, and confirmed that scaling is the right thing to do.
